framework/dynamic.go

```diff
@@ -81,65 +81,136 @@ func (frame *Framework[VertexProp, EdgeProp, MsgType]) EnactStructureChanges(g *
 			// From the gathered set of consecutive adds, apply them.
 			if len(src.OutEdges) > didxStart {
 				val := frame.AggregateRetrieve(src)
-				frame.OnEdgeAdd(g, sidx, didxStart, val)
+				msgs := frame.OnEdgeAdd(g, sidx, didxStart, val)
+				if g.Undirected || g.SendRevMsgs {
+					for eidx := didxStart; eidx < len(src.OutEdges); eidx++ {
+						edge := src.OutEdges[eidx]
+						didx := edge.Destination
+						var msg MsgType
+						if msgs != nil { // Just incase this is not needed/implemented
```

Comment:
Should we check whether msgs is null before entering the for loop?

Reply:
We enter the for loop here if the topological reverse notification is desired to be sent (either undirected graph or requested) 
This check then protects for two cases; an undirected graph (topological reverse notifications are sent, but a message value isn't required), or an implementation where the reverse notification is desired to be sent but a specific value to go along with the message is unnecessary.

Reply:
Thanks! I misread the code.